src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -465,7 +465,25 @@ private void AddDirectives(SyntaxNode node, TextSpan span)
 
                         if (_directivesToKeep.Contains(directive))
                         {
-                            AddResidualTrivia(SyntaxFactory.TriviaList(directive.ParentTrivia), requiresNewLine: true);
+                            var parentTrivia = directive.ParentTrivia;
+                            var parentToken = parentTrivia.Token;
+
+                            var triviaList = parentToken.LeadingTrivia.Contains(parentTrivia)
+                                ? parentToken.LeadingTrivia
+                                : parentToken.TrailingTrivia;
+                            var index = triviaList.IndexOf(parentTrivia);
+
+                            // If we're keeping a directive, and it's not at the start of the line, keep the whitespace
+                            // that precedes it as well.
+                            if (index >= 1 && triviaList[index - 1].Kind() == SyntaxKind.WhitespaceTrivia)
+                            {
+                                this.AddResidualTrivia(SyntaxFactory.TriviaList(
+                                    triviaList[index - 1], parentTrivia), requiresNewLine: true);
+                            }
+                            else
+                            {
+                                AddResidualTrivia(SyntaxFactory.TriviaList(parentTrivia), requiresNewLine: true);
```

Comment:
note: i'm keeping `requiresNewLine: true`.  But afaict, it's actually making for some less pleasant results.  I'm open to removing that in this PR, or having a followup for that.